packages/modules/viewer-react/src/types.ts

```diff
@@ -105,6 +106,22 @@ export interface ItwinViewerInitializerParams {
   imodelClient?: IModelClient;
 }
 
+/**
+ * list of all properties needed for iTwin.js initialization
+ * This list should be updated each time a property is added/removed from ItwinViewerInitializerParams
+ */
+export const iTwinViewerInitializerParamList = [
```

Comment:
yeah agree but I don't know of anything to extract properties from a type/interface since they don't really exist at runtime

Reply:
This could work. Thanks @MichaelBelousov. I guess my only argument is we still need to maintain a second list of the properties in the sample object.....at least they are type safe though

Reply:
if there are some default params, you can store them as an object of the Params type in this scope (if you have defaults anywhere else before, such as in conditinoal logic of the function taking the param object, you can have those reference the `defaultOptions` object you place in this scope instead of inlining them). That may allow you to logically not maintain a second list, it will just have the sideeffect of also allowing you to generate a typesafe key list. Otherwise I agree it's better to have a duplicate list for the typesafe keys in this case.

Reply:
updated